pkgs/development/python-modules/azure-mgmt-reservations/default.nix

```diff
@@ -2,40 +2,38 @@
   lib,
   buildPythonPackage,
   fetchPypi,
+  setuptools,
   msrest,
-  msrestazure,
   azure-common,
   azure-mgmt-core,
-  pythonOlder,
 }:
 
 buildPythonPackage rec {
   pname = "azure-mgmt-reservations";
   version = "2.3.0";
-  format = "setuptools";
-
-  disabled = pythonOlder "3.6";
+  pyproject = true;
 
   src = fetchPypi {
     inherit pname version;
     extension = "zip";
     hash = "sha256-BHCFEFst5jfyIEo0hm86belpxW7EygZCBJ8PTqzqHKc=";
   };
 
-  propagatedBuildInputs = [
+  build-system = [ setuptools ];
+
+  dependencies = [
     msrest
-    msrestazure
     azure-common
     azure-mgmt-core
```

Comment:
`typing-extensions` is missing, see [typing-extensions](https://github.com/Azure/azure-sdk-for-python/blob/azure-mgmt-reservations_2.3.0/sdk/reservations/azure-mgmt-reservations/setup.py#L76)

I think that for some `azure-*`modules `disabled = pythonOlder "3.9";` is set to avoid pulling in `typing-extensions`.